src/python/WMCore/MicroService/MSPileup/MSPileup.py

```diff
@@ -4,14 +4,112 @@
 Description: MSPileup provides logic behind the pileup WMCore module.
 """
 
+# system modules
+from threading import current_thread
+
 # WMCore modules
+from WMCore.REST.Auth import authz_match
 from WMCore.MicroService.MSCore.MSCore import MSCore
+from WMCore.MicroService.DataStructs.DefaultStructs import OUTPUT_REPORT
+from WMCore.MicroService.MSPileup.MSPileupData import MSPileupData
 
 
 class MSPileup(MSCore):
     """
     MSPileup provides whole logic behind the pileup WMCore module.
     """
 
-    def __init__(self, msConfig, logger=None, **kwargs):
-        super(MSPileup, self).__init__(msConfig, **kwargs)
+    def __init__(self, msConfig, **kwargs):
+        # TODO: so far we are testing this service w/o Rucio access
+        super(MSPileup, self).__init__(msConfig, skipRucio=True)
+        # TODO: if more generic approach will be required we'll need to switch to
+#         super(MSPileup, self).__init__(msConfig, **kwargs)
+
+        # for authz we should provide relevant section in MS configuration
+        # if no role/group is provided then authz_match used in HTTP APIs
+        # will always return valid state. Therefore, for test purposes we can
+        # use empty role and group, while for production setup we should set thenm up
```

Comment:
I suggest to make it last change as we still need to define proper config. So far I put basic `self.role` and `self.group` which get its values from passed config (since I do not know yet how config will look like), and I used them in `authz_match` function. If you'll provide proper config then we can adjust this.

Reply:
Giving it further thought, I would suggest to create a specific GH issue for the authorization rules and implementation. It's a requirement for deploying this service to production though. Can you please create a ticket for that?